server/commands/deployments/ProvideInfrastructure.js

```diff
@@ -89,6 +94,29 @@ function getAutoScalingTemplatesToCreate(logger, configuration, accountName, sli
   });
 }
 
+function createAsgsOnDemand(asg, configuration, slice) {
+  return createAsgInOverwriteMode(slice) ||
```

Comment:
I like splitting these out into more explicit filters but this might be better as locally scoped lambas, eg:

```
const createAsgInOverwriteMode = slice => slice === 'none';
```